third_party/nvfuser/csrc/python_frontend/python_bindings.cpp

```diff
@@ -77,6 +77,26 @@ void initNvFuserPythonBindings(PyObject* module) {
     ss << "Tensor(index=" << self.index << ", dims=" << self.dims << ")";
     return ss.str();
   });
+  tensor_class.def("__add__", [](nvfuser::Tensor& self) {});
+
+  tensor_class.def(
+      "get_fusion_definition", [](nvfuser::Tensor& self) { return self.fd; });
+
+  auto tensor_sizes = [](nvfuser::Tensor arg) -> std::vector<nvfuser::Scalar> {
```

Comment:
I'm thinking about unify tensor_sizes to a single scalar-like value.

I want to expand the concept of Scalar to a vector. something like this: https://github.com/csarofeen/pytorch/pull/2451/files#diff-df971f94c8ef070e3b8142c848e69b2d3b92054a6cf27cfc5fdc941ccf563562R76

We'll then have nested Scalar, (where a vector Scalar could hold multiple Scalar within... I know this could get messy, so I'm open to comments/suggestions as well).

Reply:
What would be the conceptual difference between that and a Tensor? Like I guess the nested version might support ragged multidimensional arrays?

Reply:
Oops, sorry about the radio silence. :stuck_out_tongue: 

It's the returned instance that I was referring to. Here we have `std::vector<nvfuser::Scalar>` as the return.
I'm thinking about changing that to a single `nvfuser::Scalar`, which will host a `std::vector<nvfuser::Scalar>`. I don't think it has any real concern for you here. Just some refactor work, so I was sort of noting this for myself.

Also, the design on that is up to discussion. Just some rough idea at this point. We can follow up when I have a PR up. :)

Reply:
Yeah, sounds good as long as we can still extract the "scalar" scalars.